lambda/update/correction.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"github.com/ministryofjustice/opg-data-lpa-store/internal/shared"
+	"github.com/ministryofjustice/opg-data-lpa-store/internal/validate"
+	"github.com/ministryofjustice/opg-data-lpa-store/lambda/update/parse"
+	"time"
+)
+
+type Correction struct {
+	DonorFirstNames string
+	DonorLastName   string
+	DonorOtherNames string
+	DonorDob        shared.Date
+	DonorAddress    shared.Address
+	DonorEmail      string
+	LPASignedAt     time.Time
+}
+
+func (c Correction) Apply(lpa *shared.Lpa) []shared.FieldError {
+
+	lpa.Donor.FirstNames = c.DonorFirstNames
+	lpa.Donor.LastName = c.DonorLastName
+	lpa.Donor.OtherNamesKnownBy = c.DonorOtherNames
+	lpa.Donor.DateOfBirth = c.DonorDob
+	lpa.Donor.Address = c.DonorAddress
+	lpa.Donor.Email = c.DonorEmail
+	lpa.SignedAt = c.LPASignedAt
```

Comment:
If [this requirement](https://opgtransform.atlassian.net/browse/VEGA-2667?focusedCommentId=95756) in the ticket comments is still true we may want to add some protection here to check if it is a paper LPA.

Reply:
It's meant to be pushed to another ticket which hasn't been written up yet, but I've added it in.